tests/NLog.Web.AspNetCore.Tests/NLogBufferingTargetWrapperMiddlewareTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using NLog.Web.Internal;
+using NLog.Web.Targets.Wrappers;
+using Xunit;
+
+namespace NLog.Web.Tests
+{
+    public class NLogBufferingTargetWrapperMiddlewareTests
+    {
+        private static LogFactory RegisterAspNetCoreBufferingTargetWrapper(string targetName)
+        {
+            LogManager.LogFactory.Setup().RegisterNLogWeb().LoadConfigurationFromXml(
+                $@"<nlog throwConfigExceptions='true'>
+                    <targets>
+                        <target 
+                            type='AspNetBufferingWrapper' 
+                            name='{targetName}' 
+                            bufferGrowLimit='9' 
+                            bufferSize='7' 
+                            growBufferAsNeeded='true'>
+                            <target type='Memory' name='unitTestMemoryTarget' />
+                        </target>
+                    </targets>
+                    <rules>
+                        <logger name='*' minlevel='Debug' writeTo='{targetName}' />
+                    </rules>
+                </nlog>");
+            return LogManager.LogFactory;
+        }
+
+        [Fact]
+        public async Task BufferingMiddlewareInvokeTest()
+        {
+            var logFactory = RegisterAspNetCoreBufferingTargetWrapper("first");
+
+            Assert.NotNull(logFactory?.Configuration?.FindTargetByName<AspNetBufferingTargetWrapper>("first"));
+
+            DefaultHttpContext context = new DefaultHttpContext();
+
+            var target = logFactory?.Configuration?.FindTargetByName<AspNetBufferingTargetWrapper>("first");
+            target.HttpContextAccessor = new FakeHttpContextAccessor(context);
+
+            RequestDelegate next = (HttpContext hc) =>
+            {
+                ILogger logger = logFactory.GetCurrentClassLogger();
+
+                for(int i = 0; i < 10; i++)
+                {
+                    logger.Debug("This is a unit test logging.");
+                }
+
+                return Task.CompletedTask;
+            };
+
+            NLogBufferingTargetWrapperMiddleware middleware = new NLogBufferingTargetWrapperMiddleware(next);
+
+
+
+            await middleware.Invoke(context).ConfigureAwait(false);
+
+            Assert.NotEmpty(context.Items);
```

Comment:
Instead of doing white-box Assert of the inner-details inside `context.Items`, then I think the unit-test should focus on checking `unitTestMemoryTarget`:
- Logging without having active HttpContext will make it forward directly to `unitTestMemoryTarget`
- Logging within active HttpContext will not reach `unitTestMemoryTarget` until after `OnEndRequest`